huaweicloud/services/dws/resource_huaweicloud_dws_workload_queue.go

```diff
@@ -0,0 +1,235 @@
+		httpUrl = "v2/{project_id}/clusters/{cluster_id}/workload/queues"
+		product = "dws"
+	)
+
+	client, err := cfg.NewServiceClient(product, region)
+	if err != nil {
+		return diag.Errorf("error creating DWS client: %s", err)
+	}
+
+	createPath := client.Endpoint + httpUrl
+	createPath = strings.ReplaceAll(createPath, "{project_id}", client.ProjectID)
+	createPath = strings.ReplaceAll(createPath, "{cluster_id}", d.Get("cluster_id").(string))
+	createOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+		JSONBody:         utils.RemoveNil(buildCreateWorkloadQueueBodyParams(d)),
+	}
+
+	_, err = client.Request("PUT", createPath, &createOpt)
+	if err != nil {
+		return diag.Errorf("error creating DWS workload queue: %s", err)
+	}
+
+	d.SetId(d.Get("name").(string))
+
+	return resourceWorkLoadQueueRead(ctx, d, meta)
+}
+
+func buildCreateWorkloadQueueBodyParams(d *schema.ResourceData) map[string]interface{} {
+	return map[string]interface{}{
+		"workload_queue": map[string]interface{}{
+			"workload_queue_name":         d.Get("name"),
+			"workload_resource_item_list": buildCreateConfigurationBodyParams(d),
+			"logical_cluster_name":        utils.ValueIngoreEmpty(d.Get("logical_cluster_name")),
+		},
+	}
+}
+
+func buildCreateConfigurationBodyParams(d *schema.ResourceData) []map[string]interface{} {
+	rawParams := d.Get("configurations").(*schema.Set)
+	if rawParams.Len() == 0 {
+		return nil
+	}
+
+	params := make([]map[string]interface{}, 0, rawParams.Len())
+	for _, rawParam := range rawParams.List() {
+		raw := rawParam.(map[string]interface{})
+		param := map[string]interface{}{
+			"resource_name":  raw["resource_name"],
+			"resource_value": raw["resource_value"],
+		}
+		params = append(params, param)
+	}
+
+	return params
+}
+
+func resourceWorkLoadQueueRead(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	var (
+		cfg     = meta.(*config.Config)
+		region  = cfg.GetRegion(d)
+		httpUrl = "v2/{project_id}/clusters/{cluster_id}/workload/queues"
+		product = "dws"
+	)
+
+	getClient, err := cfg.NewServiceClient(product, region)
+	if err != nil {
+		return diag.Errorf("error creating DWS client: %s", err)
+	}
+
+	getPath := getClient.Endpoint + httpUrl
+	getPath = strings.ReplaceAll(getPath, "{project_id}", getClient.ProjectID)
+	getPath = strings.ReplaceAll(getPath, "{cluster_id}", d.Get("cluster_id").(string))
+	getOpt := golangsdk.RequestOpts{
+		KeepResponseBody: true,
+	}
+
+	getResp, err := getClient.Request("GET", getPath, &getOpt)
+	if err != nil {
+		return common.CheckDeletedDiag(d, err, "DWS workload queue")
+	}
+
+	getRespBody, err := utils.FlattenResponse(getResp)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	resp := utils.PathSearch("workload_queue_name_list", getRespBody, make([]interface{}, 0))
+	if !containsResourceID(resp, d.Id()) {
+		return common.CheckDeletedDiag(d, golangsdk.ErrDefault404{}, "DWS workload queue")
+	}
+
+	// Due to API restrictions, only return a list of names.
```

Comment:
```suggestion
	// Due to API restrictions, only the queue name can be found in the API response.
```
Comments here should not mention list, because of the list is handled in the code above

Reply:
fixed.